debezium-connector-mysql/src/test/java/io/debezium/connector/mysql/junit/SkipTestDependingOnDatabaseVersionRule.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.connector.mysql.junit;
+
+import org.junit.runner.Description;
+import org.junit.runners.model.Statement;
+
+import io.debezium.connector.mysql.MySQLConnection;
+import io.debezium.connector.mysql.MySQLConnection.MySqlVersion;
+import io.debezium.junit.AnnotationBasedTestRule;
+
+/**
+ * JUnit rule that skips a test based on the {@link SkipWhenDatabaseVersion} annotation either on a test method or
+ * on a test class.
+ *
+ * @author Chris Cranford
+ */
+public class SkipTestDependingOnDatabaseVersionRule extends AnnotationBasedTestRule {
```

Comment:
We have similar rule in postgres. Would it it make sense to provide it in `core` test and have only the annotations defined per-connector?

Reply:
Do I understand you correctly that the annotations would be connector-specific but would also live in the core test suite; otherwise I'm unclear how that would ultimately work.

One idea I had however was to introduce an interface that the rule would load any impl of that was on the classpath and this interface would be responsible for returning the source major, minor, and patch version values.  We'd then have generalized annotations like:

`@SkipWhenDatabaseVersionLessThan(...)`
`@SkipWhenDatabaseVersionLessThanOrEqualTo(...)`
`@SkipWhenDatabaseVersionEqualTo(...)`
`@SkipWhenDatabaseVersionGreaterThanOrEqualTo(...)`
`@SkipWhenDatabaseVersionGreaterThan(...)`

All these would allow for specifying a `major`, `minor`, and `patch` value along with `reason` to describe why the test is to be skipped.  How does that sound?

Reply:
Yeah, that's roughly my reasoning. And I must admit I like  the second proposal even more!